.github/workflows/java-release.yml

```diff
@@ -8,13 +8,196 @@ on:
         description: "The Dafny version to run"
         required: true
         type: string
+      version:
+        description: "The version being released"
+        required: true
+        type: string
+  # Useful to run this action by itself since sometimes to apply hot patches
+  # to just java we need to only release a jva build
+  workflow_dispatch:
+    inputs:
+      dafny:
+        description: "The Dafny version to run"
+        required: true
+        type: string
+      version:
+        description: "The version being released"
+        required: true
+        type: string
 
 jobs:
-  releaseJava:
+  allowList:
+    uses: ./.github/workflows/validate_user.yml
+  releaseToCA:
+    needs: allowList
+    strategy:
+      matrix:
+        library: [AwsCryptographicMaterialProviders]
+        dotnet-version: ["6.0.x"]
+    runs-on: codebuild-AWS-MPL-Release-${{ github.run_id }}-${{ github.run_attempt }}-ubuntu-5.0-large
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      REGION: us-east-1
+      DOMAIN: crypto-tools-internal
+      REPOSITORY: java-mpl-staging
+      CODEARTIFACT_REPO_URL: https://crypto-tools-internal-587316601012.d.codeartifact.us-east-1.amazonaws.com/maven/java-mpl-staging/
+    steps:
+      - name: Support longpaths on Git checkout
+        run: |
+          git config --global core.longpaths true
+
+      - uses: actions/checkout@v3
+      # We only pull in the submodules we need to build the library
+      - run: git submodule update --init libraries
+
+      - name: Retrieve User and Password for Signing Keys in SecretsManager
+        uses: aws-actions/aws-secretsmanager-get-secrets@v2
+        with:
+          secret-ids: arn:aws:secretsmanager:us-west-2:587316601012:secret:Maven-GPG-Keys-Release-Credentials-WgJanS
+          parse-json-secrets: true
+
+      - name: Retrieve Keys from SecretsManager and add keys to KeyChain
+        run: |
+          sudo apt-get install gnupg2 -y
+          aws secretsmanager get-secret-value --region us-west-2 --secret-id Maven-GPG-Keys-Release --query SecretBinary --output text | base64 -d > ~/mvn_gpg.tgz
+          tar -xvf ~/mvn_gpg.tgz -C ~
+          mkdir /root/.gnupg
+          printf "use-agent\npinentry-mode loopback" >> ~/mvn_gpg/gpg.conf
+          printf "allow-loopback-pinentry" >> ~/mvn_gpg/gpg-agent.conf
+          mv -v ~/mvn_gpg/* /root/.gnupg/
```

Comment:
Question: When this is run, who can see the GHW logs?
Can we restrict to just "us"?

Reply:
I don't think you can restrict who can see the GHW log, but why would we? Do we care? All secrets are abstracted in the log so GH will not let you log them.
In either case, for this step we are not printing any creds or anything crazy.